java/src/main/java/org/rocksdb/WriteBatchWithIndex.java

```diff
@@ -152,8 +169,13 @@ public RocksIterator newIteratorWithBase(
    * @return An iterator which shows a view comprised of both the database
    * point-in-time from baseIterator and modifications made in this write batch.
    */
-  public RocksIterator newIteratorWithBase(final ColumnFamilyHandle columnFamilyHandle,
-      final RocksIterator baseIterator, /* @Nullable */ final ReadOptions readOptions) {
+  public /* @Nullable */ RocksIterator newIteratorWithBase(
+      final ColumnFamilyHandle columnFamilyHandle, final RocksIterator baseIterator,
+      /* @Nullable */ final ReadOptions readOptions) {
+    if (!overwriteKey) {
+      return null;
+    }
```

Comment:
Instead of this, why not call iteratorWithBase and return null if the JNI method returned 0 (effectively null).  That way, if the underlying C++ code changes, the Java should "just work"

Reply:
Every call from Java into C++ is expensive, so we try and avoid them wherever possible.